diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0789c20..b357fd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ string(COMPARE EQUAL ${CMAKE_SYSTEM_NAME} "Linux" IS_LINUX)

 # Project options
 include(CMakeDependentOption)
-cmake_dependent_option(PLUME_SDL_VULKAN_ENABLED "Enable SDL Vulkan integration" OFF IS_LINUX OFF)
+cmake_dependent_option(PLUME_SDL_VULKAN_ENABLED "Enable SDL Vulkan integration" OFF "IS_LINUX OR ANDROID" OFF)
 cmake_dependent_option(PLUME_D3D12_AGILITY_SDK_ENABLED "Enable D3D12 Agility SDK" OFF WIN32 OFF)
 cmake_dependent_option(PLUME_APPLE_RETINA_ENABLED "Enable Apple Retina display support" ON APPLE ON)
 option(PLUME_BUILD_EXAMPLES "Build example applications" OFF)
diff --git a/plume_render_interface_types.h b/plume_render_interface_types.h
index 9dcfedf..d6153a5 100644
--- a/plume_render_interface_types.h
+++ b/plume_render_interface_types.h
@@ -37,10 +37,10 @@ namespace plume {
 #if defined(_WIN64)
     // Native HWND handle to the target window.
     typedef HWND RenderWindow;
-#elif defined(__ANDROID__)
-    typedef ANativeWindow* RenderWindow;
 #elif defined(PLUME_SDL_VULKAN_ENABLED)
     typedef SDL_Window *RenderWindow;
+#elif defined(__ANDROID__)
+    typedef ANativeWindow* RenderWindow;
 #elif defined(__linux__)
     struct RenderWindow {
         Display* display;
diff --git a/plume_vulkan.cpp b/plume_vulkan.cpp
index 2800022..319a53e 100644
--- a/plume_vulkan.cpp
+++ b/plume_vulkan.cpp
@@ -4458,7 +4458,13 @@ namespace plume {
 #else
     VulkanInterface::VulkanInterface() {
 #endif
+#if defined(__ANDROID__) && defined(PLUME_SDL_VULKAN_ENABLED)
+        auto getProc = reinterpret_cast<PFN_vkGetInstanceProcAddr>(SDL_Vulkan_GetVkGetInstanceProcAddr());
+        VkResult res = getProc ? VK_SUCCESS : VK_ERROR_INITIALIZATION_FAILED;
+        if (getProc) volkInitializeCustom(getProc);
+#else
         VkResult res = volkInitialize();
+#endif
         if (res != VK_SUCCESS) {
             fprintf(stderr, "volkInitialize failed with error code 0x%X.\n", res);
             return;
